cirq-core/cirq/sim/sparse_simulator_test.py

```diff
@@ -1434,3 +1434,22 @@ def test_unseparated_states_str():
 qubits: (cirq.LineQubit(0), cirq.LineQubit(1))
 output vector: 0.707j|00⟩ + 0.707j|10⟩"""
     )
+
+
+@pytest.mark.parametrize('split', [True, False])
+def test_measurement_preserves_phase(split: bool):
+    c1, c2, t = cirq.LineQubit.range(3)
+    circuit = cirq.Circuit(
+        cirq.H(t),
+        cirq.measure(t, key="t"),
+        cirq.CZ(c1, c2).with_classical_controls("t"),
+        cirq.reset(t),
+    )
+    simulator = cirq.Simulator(split_untangled_states=split)
+    # Run enough times that both options of |110> - |111> are likely measured.
+    for _ in range(20):
+        for step in simulator.simulate_moment_steps(
+            circuit, initial_state=(1, 1, 1), qubit_order=(c1, c2, t)
+        ):
+            pass
```

Comment:
We don't need to simulate moment by moment to test the behavior. Directly using `simulator.simulate(circuit, initial_state=(1, 1, 1)).dirac_notation()` should also work.